### ISSUE TITLE
Pose optimization for 2dgs?

### ISSUE THREAD

**Issue by jizongFox** [nerfstudio-project/gsplat#546]
Hello,
I happened to compare the pose optimization gradient between 3dgs and 2dgs. I found that viewmat has always 0.0 gradient. A minimal example can be found as blew.

```python
import torch

from gsplat import rasterization_2dgs

device = torch.device("cuda:0")

means = torch.randn((1000, 3), device=device, requires_grad=True)
quats = torch.randn((1000, 4), device=device, requires_grad=True)
scales = torch.rand((1000, 3), device=device, requires_grad=True) * 0.1
colors = torch.rand((1000, 3), device=device, requires_grad=True)
opacities = torch.rand((1000,), device=device, requires_grad=True)
# define cameras
viewmats = torch.eye(4, device=device)[None, :, :]
viewmats.requires_grad = True
Ks = torch.tensor(
    [[300.0, 0.0, 150.0], [0.0, 300.0, 100.0], [0.0, 0.0, 1.0]],
    device=device,
    requires_grad=True,
)[None, :, :]
width, height = 300, 200
# render
colors, alphas, normals, surf_normals, distort, median_depth, meta = (
    rasterization_2dgs(
        means, quats, scales, opacities, colors, viewmats, Ks, width, height
    )
)
loss = colors.mean()
loss.backward()

print("Gradients of means:", means.grad)
print("Gradients of quats:", quats.grad)
print("Gradients of scales:", scales.grad)
print("Gradients of colors:", colors.grad)
print("Gradients of opacities:", opacities.grad)
print("Gradients of viewmats:", viewmats.grad.max())
print("Gradients of Ks:", Ks.grad)


```
What we have observed was 
```bash
Gradients of viewmats: tensor(0., device='cuda:0')
```
While this value is very large with 3dgs. 

I am wondering if this is designed.

**Comment by maturk** [comment]
Good point. @FantasticOven2 @liruilong940607, do you guys know how to enable pose optimization for 2DGS?

**Comment by maturk** [comment]
It seems that an equivalent computation of `v_viewmats` (3DGS equivalent [here](https://github.com/nerfstudio-project/gsplat/blob/c8f048f25c455ddd54b437ef04e32bd4e9e61f25/gsplat/cuda/csrc/ProjectionEWA3DGSFused.cu#L513)) is missing for the 2DGS projection.

**Comment by FantasticOven2** [comment]
We added viewmats gradient for 2dgs [here](https://github.com/nerfstudio-project/gsplat/pull/645)!
